cros_gralloc/cros_gralloc_driver.cc

```diff
@@ -109,6 +114,83 @@ cros_gralloc_driver *cros_gralloc_driver::get_instance()
         close(fd);                   \
     }
 
+int32_t cros_gralloc_driver::reload()
+{
+	int fd;
+	drmVersionPtr version;
+	char const *str = "%s/renderD%d";
+	uint32_t j;
+	char *node;
+
+	const int render_num = 10;
+	const int name_length = 50;
+	int availabe_node = 0;
+
+	availabe_node = 0;
+	for (uint32_t i = drv_num_; i < drv_num_ + IVSH_DEVICE_NUM; i++) {
+		if (asprintf(&node, str, DRM_DIR_NAME, i) < 0)
+			continue;
+
+		fd = open(node, O_RDWR, 0);
+		free(node);
+		if (fd < 0)
+			continue;
+
+		version = drmGetVersion(fd);
+		if (!version) {
+			close(fd);
+			continue;
+		}
+
+		availabe_node++;
+		drv_num_ = i;
+		close(fd);
+		drmFreeVersion(version);
+	}
+
+	// have the ivshemem node
+	if ((availabe_node > 0) && !drv_ivshmem_) {
+		if (asprintf(&node, str, DRM_DIR_NAME, drv_num_) < 0) {
+			drv_loge("Open new added node fail\n");
+			return -ENODEV;
+		}
+
+		fd = open(node, O_RDWR, 0);
```

Comment:
need check fd, here, upper log
drv_loge("Open new added node fail\n");
need be moved here

Reply:
in the line 136, checked the fd, the availabe_node calculate the available node, in line 152, availabe_node > 0 means has the available node, so the fd should be the workable fd.
I add the checks before, Chenyan comment that the checks are done above, so i remove the checks.

Reply:
Actually what i mean is that u can push this bloc code(line 163-190) into the for loop (line 145), and rm line 152-162. Just for reference.

Reply:
agree, suggest to combine upper code to remove some duplicated code. Also please help add some comments here, for example, the second ivshmem device is for our screen cast purpose.
Also please add enough introduction in the commit description.

Reply:
Patch updated, i made changes to use the first available ivshm node to screen cast.
Please check.